NDTensors/ext/NDTensorscuTENSORExt/contract.jl

```diff
@@ -1,7 +1,12 @@
-using NDTensors: NDTensors, DenseTensor, array
+using NDTensors: NDTensors, DenseTensor, array, data
 using NDTensors.Expose: Exposed, unexpose
 using cuTENSOR: cuTENSOR, CuArray, CuTensor
 
+struct CuArrayOffset{N} end
+function CuArrayOffset(V)
```

Comment:
@mtfishman  I made this struct to be a trait for `contract!` launch so we don't always copy the `CuArrays`. In the copy based `contract!` function I copy all of the `CuArrays` but I could see it where I create multiple functions where I only copy the necessary `CuArrays`. Would this more refined approach be preferred or is it okay to copy them all for now.